### PR TITLE
Apply `clang-tidy` suggestions to tutorials

### PR DESCRIPTION
**Pull request by guitargeek**
For the tutorials, this is good to do in particular, since they are also examples for good coding practices to the users

### CODE REVIEW

File: tutorials/math/mathcoreGenVector.C [hunk 1]
```diff
@@ -380,10 +380,10 @@ int testLorentzVector() {
...
    // test operator ==
-   XYZTVector        w1 = v1;
-   PtEtaPhiEVector   w2 = v2;
-   PtEtaPhiMVector   w3 = v3;
-   PxPyPzMVector     w4 = v4;
+   const XYZTVector&        w1 = v1;
+   const PtEtaPhiEVector&   w2 = v2;
+   const PtEtaPhiMVector&   w3 = v3;
+   const PxPyPzMVector&     w4 = v4;
    ok+= compare( w1 == v1, static_cast<double>(true), "== PxPyPzE");
```
Comment (linev) [inline tutorials/math/mathcoreGenVector.C:387]: I think, here assign operator also was tested.
Therefore modified code is not equivalent to previous one.

File: tutorials/unfold/testUnfold7c.C [hunk 2]
```diff
@@ -1512,7 +1512,7 @@ TFitResultPtr DoFit(TH1 *h,TH2 *rho,TH1 *truth,const char *text,
             exit(0);
          }
       }
-      TMatrixD O(ev.GetEigenVectors());
+      const TMatrixD& O(ev.GetEigenVectors());
       TMatrixD DOT(d,TMatrixD::kMultTranspose,O);
```
Comment (linev) [inline tutorials/unfold/testUnfold7c.C:1516]: Same is here. Previous version made a copy of matrix. Now is just reference.

File: tutorials/unfold/testUnfold7c.C [hunk 3]
```diff
@@ -1552,7 +1552,7 @@ TFitResultPtr DoFit(TH1 *h,TH2 *rho,TH1 *truth,const char *text,
             exit(0);
          }
       }
-      TMatrixD O1(ev1.GetEigenVectors());
+      const TMatrixD& O1(ev1.GetEigenVectors());
       TMatrixD DOT1(d1,TMatrixD::kMultTranspose,O1);
```
Comment (linev) [inline tutorials/unfold/testUnfold7c.C:1556]: Also here

**Review by linev — CHANGES_REQUESTED** [review]
One should not make reference on the objects but keep copy constructors.

**Comment by guitargeek** [comment]
Thank you Serguei! I should have removed the corresponding `clang-tidy` check from the list.

Can you take a look again? I will squash the commits on merge.

**Review by linev — APPROVED** [review]

File: tutorials/math/mathcoreGenVector.C (hunk 1 above)
Comment (pcanal) [reply tutorials/math/mathcoreGenVector.C]: One probably should separate the testing code from the tutorials (which should show user the right way to use the classes)

**Pull request merged**